CMSgov/easi-app
# [EASI-4129] nginx CEDAR Caching

# EASI-4129, EASI-4130

## Changes and Description

- Implements nginx as a proxy between EASi and CEDAR that caches all GET requests.
- Implements a community module for nginx that allows for cache invalidation by sending an HTTP PURGE request to any cached endpoint's URL.

## How to test this change

Connect to the CMS VPN, start the app using docker-compose as normal, and make sure to set `.envrc.local` to allow connecting to CEDAR. Requests should cache automatically. Adding/removing users/roles to systems has cache invalidation applied as an example.

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
